akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -438,19 +436,22 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
       log.debug("Started embedded media driver in directory [{}]", driver.aeronDirectoryName)
       topLevelFREvents.loFreq(Transport_MediaDriverStarted, driver.aeronDirectoryName().getBytes("US-ASCII"))
       Runtime.getRuntime.addShutdownHook(stopMediaDriverShutdownHook)
-      mediaDriver = Some(driver)
+      mediaDriver.set(Some(driver))
```

Comment:
You only get here through `ArteryTransport.start` and I'm pretty sure there will all kinds of catastrophy that already happened if you do that more than once.